syntax = "proto3";

package bayes.bob;

message Feedback {
  // ID of the user that sent the feedback.
  string user_id = 1;

  // ID of the project for which the feedback was sent.
  string project_id = 2;

  // ID of the advice module for which the feedback was sent.
  string advice_id = 3;

  // Actual feedback.
  string feedback = 4;

  // Source of the feedback.
  FeedbackSource source = 5;

  // Numerical score (usually between 1 and 5 but may vary depending on the source).
  int32 score = 6;
}

enum FeedbackSource {
  UNKNOWN_SOURCE = 0;

  // Feedback on a specific advice.
  ADVICE_FEEDBACK = 1;

  // General feedback in the "professionals" page.
  PROFESSIONAL_PAGE_FEEDBACK = 2;

  // Global feedback on a project.
  PROJECT_FEEDBACK = 3;

  // Feedback given in the Changelog modals (on specific questions).
  CHANGELOG_FEEDBACK = 4;

  // Suggest a new advice.
  NEW_ADVICE_FEEDBACK = 5;

  // General comment on the product.
  PRODUCT_FEEDBACK = 6;
}
